src/cpp/core/include/core/system/System.hpp

```diff
@@ -342,7 +343,40 @@ std::vector<SubprocInfo> getSubprocesses(PidType pid);
 // if unable to determine cwd
 FilePath currentWorkingDir(PidType pid);
 
+struct ProcessInfo
+{
+   ProcessInfo() : pid(0), ppid(0), pgrp(0) {}
+   PidType pid;
+   PidType ppid;
+   PidType pgrp;
+   std::string username;
+   std::string exe;
+   std::string state;
+   std::vector<std::string> arguments;
+
+#if !defined _WIN32 && !defined __APPLE__
+   core::Error creationTime(boost::posix_time::ptime* pCreationTime) const;
+#endif
+};
+
+// simple encapsulation of parent-child relationship of processes
+struct ProcessTreeNode
+{
+   boost::shared_ptr<ProcessInfo> data;
+   std::vector<boost::shared_ptr<ProcessTreeNode> > children;
```

Comment:
Nit: in C++ 11 I believe the space between `> >` is no longer necessary?

Reply:
Good to know - I was wondering why removing the space seemed to start working all of a sudden when I distinctly remember it used to not some months ago.